ui_testing/testcases/basic_tests/test006_orders.py

```diff
@@ -2871,3 +2871,30 @@ def test084_filter_by_changed_by(self):
             self.assertTrue(key_found)
             # close child table
             self.orders_page.close_child_table(source=results[i])
+
+    def test090_if_cancel_archive_order_no_order_suborder_analysis_will_archived(self):
+        """
+        [Archiving][MainOrder]Make sure that if user cancel archive order,
+        No order or suborders or analysis of the order will be archived
+        LIMS-5404
+        """
+        self.info('create order')
+        self.order_page.create_new_order(material_type='Raw Material', save=False)
+        self.info('dupliacte the suborder for 2 times')
+        self.order_page.duplicate_from_table_view(number_of_duplicates=2)
+        self.order_page.save(save_btn='order:save_btn')
+        order_no = self.order_page.get_no()
+        order_id = self.order_page.get_order_id()
+        suborders_data, _ = self.orders_api.get_suborder_by_order_id(order_id)
+        suborders = suborders_data['orders']
+        self.assertEqual(3, len(suborders))
+        self.orders_page.get_orders_page()
+        self.orders_page.filter_by_order_no(filter_text=order_no)
+        self.info('click on arhcive then cancel popup')
+        self.orders_page.archive_main_order_from_order_option(check_pop_up=True, confirm=False)
+        table_records = self.orders_page.result_table(element='general:table')
+        self.assertEqual(1, len(table_records) - 1)
+        self.info('go to archived orders')
+        self.orders_page.get_archived_items()
+        self.orders_page.search(order_no)
```

Comment:
Replace search with filter by order no.. As search gets all data with this no may be analysis has same no of your order.

Reply:
done